### ISSUE TITLE
Instrument name on part is not dynamically determined

### ISSUE THREAD

**Issue by VictorEijkhout** [musescore/MuseScore#19424]
### Issue type

UI bug

### Bug description

Changing the name of a part does not alter the instrument name on the part

### Steps to reproduce

1. Create a part
2. give it a non-obvious name, say "X"
3. open the part: it will have "X" as instrument name
4. go to the parts dialog and change part name to "Y"
5. part name is changed in the tab, but not in the part
6.

### MuseScore Version

4.1.1

### Regression

I don't know

### Operating system

MacOS 13

**Comment by XiaoMigros** [comment]
This is completely intentional- Often, you will want a part name that doesn't match the instrument name, especially if your parts contain more than one instrument

**Comment by bkunda** [comment]
Actually, the part name _does_ update.
You just have to double click on it to prompt it to do so...(which is probably a bit weird. If we support this, then it would be nicer if it just happened automatically). 

https://github.com/musescore/MuseScore/assets/86290556/83793963-d48d-4251-852b-0055d8e4002b

@oktophonie @mike-spa @miiizen FYI

**Comment by VictorEijkhout** [comment]
Yes, I noticed this later. Even stranger, you only have to click on *one* part's instrument name for *all* of them to suddenly update.
